lib/Sema/CSDiag.cpp

```diff
@@ -2315,16 +2315,22 @@ bool FailureDiagnosis::diagnoseContextualConversionError(
       exprType->isEqual(contextualType)) {
     return false;
   }
-  
+
   // If we're trying to convert something of type "() -> T" to T, then we
   // probably meant to call the value.
   if (auto srcFT = exprType->getAs<AnyFunctionType>()) {
     if (srcFT->getParams().empty() &&
         !isUnresolvedOrTypeVarType(srcFT->getResult()) &&
         CS.TC.isConvertibleTo(srcFT->getResult(), contextualType, CS.DC)) {
-      diagnose(expr->getLoc(), diag::missing_nullary_call, srcFT->getResult())
-        .highlight(expr->getSourceRange())
-        .fixItInsertAfter(expr->getEndLoc(), "()");
+
+      auto locator = CS.getConstraintLocator(expr);
+      auto extendedLocator =
+          CS.getConstraintLocator(locator, ConstraintLocator::ContextualType);
```

Comment:
Nit: you can do `CS.getConstraintLocator(expr,  ConstraintLocator::ContextualType);`

Reply:
Done!